api/v1/server/handlers/monitoring/probe.go

```diff
@@ -0,0 +1,320 @@
+package monitoring
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"slices"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/labstack/echo/v4"
+	"golang.org/x/exp/rand"
+
+	"github.com/hatchet-dev/hatchet/api/v1/server/oas/gen"
+	"github.com/hatchet-dev/hatchet/pkg/client"
+	"github.com/hatchet-dev/hatchet/pkg/cmdutils"
+	clientconfig "github.com/hatchet-dev/hatchet/pkg/config/client"
+	"github.com/hatchet-dev/hatchet/pkg/config/shared"
+	"github.com/hatchet-dev/hatchet/pkg/repository/prisma/db"
+	"github.com/hatchet-dev/hatchet/pkg/repository/prisma/dbsqlc"
+	"github.com/hatchet-dev/hatchet/pkg/repository/prisma/sqlchelpers"
+	"github.com/hatchet-dev/hatchet/pkg/worker"
+)
+
+func (m *MonitoringService) MonitoringPostRunProbe(ctx echo.Context, request gen.MonitoringPostRunProbeRequestObject) (gen.MonitoringPostRunProbeResponseObject, error) {
+	if !m.enabled {
+		return nil, fmt.Errorf("monitoring is not enabled")
+	}
+
+	tenant := ctx.Get("tenant").(*db.TenantModel)
+
+	if !slices.Contains[[]string](m.permittedTenants, tenant.ID) {
+
+		err := fmt.Errorf("tenant is not a monitoring tenant for this instance")
+
+		if len(m.permittedTenants) > 0 {
+			m.l.Error().Err(err).Msgf("monitoring tenants are %v", m.permittedTenants)
+		} else {
+			m.l.Error().Err(err).Msg("no monitoring tenants are configured")
+		}
+
+		return gen.MonitoringPostRunProbe403JSONResponse{}, nil
+	}
+
+	token, err := getBearerTokenFromRequest(ctx.Request())
+
+	if err != nil {
+		return nil, err
+	}
+
+	events := make(chan string, 50)
+	stepChan := make(chan string, 50)
+	errorChan := make(chan error, 50)
+	interrupt := cmdutils.InterruptChan()
+
+	cancellableContext, cancel := context.WithTimeout(ctx.Request().Context(), m.probeTimeout)
+
+	defer cancel()
+	var cleanup func() error
+
+	cf := clientconfig.ClientConfigFile{
+		Token:     token,
+		TenantId:  tenant.ID,
+		Namespace: randomNamespace(),
+		TLS: clientconfig.ClientTLSConfigFile{
+			Base: shared.TLSConfigFile{
+				TLSRootCAFile: m.tlsRootCAFile,
+			},
+		},
+	}
+
+	wfrId, cleanup, err := m.run(cancellableContext, cf, m.workflowName, events, stepChan, errorChan)
+	if err != nil {
+		m.l.Error().Msgf("error running probe: %s", err)
+		return nil, err
+	}
+
+	defer func() {
```

Comment:
To make this more readable, I think we can write this as `defer cleanup()` and:

1. The callee (`m.run`) should ensure that if `err == nil`, then `cleanup` is never nil
2. Errors should be logged from within the cleanup method itself